tests/unit/s2n_ktls_io_test.c

```diff
@@ -523,5 +599,242 @@ int main(int argc, char **argv)
+            EXPECT_OK(s2n_test_records_in_ancillary(&out, 0));
+
+            /* Send nothing with iovec array with zero-length buffer */
+            EXPECT_EQUAL(s2n_ktls_sendv_with_offset(conn, &test_iovec, 1, 0, &blocked), 0);
+            EXPECT_EQUAL(out.sendmsg_invoked_count, 2);
+            EXPECT_EQUAL(blocked, S2N_NOT_BLOCKED);
+            EXPECT_OK(s2n_test_records_in_ancillary(&out, 0));
+        };
+
+        /* Test: Send with multiple iovecs */
+        {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(conn);
+
+            DEFER_CLEANUP(struct s2n_test_ktls_io_stuffer out = { 0 },
+                    s2n_ktls_io_stuffer_free);
+            EXPECT_OK(s2n_test_init_ktls_io_stuffer_send(conn, &out));
+
+            DEFER_CLEANUP(struct s2n_test_iovecs test_iovecs = { 0 }, s2n_test_iovecs_free);
+            EXPECT_OK(s2n_test_new_iovecs(&test_iovecs, &test_data_blob,
+                    test_iov_lens, s2n_array_len(test_iov_lens)));
+
+            s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+            ssize_t result = s2n_ktls_sendv_with_offset(conn,
+                    test_iovecs.iovecs, test_iovecs.iovecs_count, 0, &blocked);
+            EXPECT_EQUAL(result, sizeof(test_data));
+
+            EXPECT_EQUAL(out.sendmsg_invoked_count, 1);
+            EXPECT_EQUAL(blocked, S2N_NOT_BLOCKED);
+            EXPECT_OK(s2n_test_validate_ancillary(&out, TLS_APPLICATION_DATA, sizeof(test_data)));
+            EXPECT_OK(s2n_test_validate_data(&out, test_data, sizeof(test_data)));
+        };
+
+        /* Test: Send with very large number of iovecs */
+        {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(conn);
+
+            DEFER_CLEANUP(struct s2n_test_ktls_io_stuffer out = { 0 },
+                    s2n_ktls_io_stuffer_free);
+            EXPECT_OK(s2n_test_init_ktls_io_stuffer_send(conn, &out));
+
+            size_t many_iov_lens[1000] = { 0 };
+            DEFER_CLEANUP(struct s2n_test_iovecs test_iovecs = { 0 }, s2n_test_iovecs_free);
+            EXPECT_OK(s2n_test_new_iovecs(&test_iovecs, &test_data_blob,
+                    many_iov_lens, s2n_array_len(many_iov_lens)));
+
+            s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+            ssize_t result = s2n_ktls_sendv_with_offset(conn,
+                    test_iovecs.iovecs, test_iovecs.iovecs_count, 0, &blocked);
+            EXPECT_EQUAL(result, sizeof(test_data));
+
+            EXPECT_EQUAL(out.sendmsg_invoked_count, 1);
+            EXPECT_EQUAL(blocked, S2N_NOT_BLOCKED);
+            EXPECT_OK(s2n_test_validate_ancillary(&out, TLS_APPLICATION_DATA, sizeof(test_data)));
+            EXPECT_OK(s2n_test_validate_data(&out, test_data, sizeof(test_data)));
+        };
+
+        /* Test: Send with offset */
```

Comment:
Is there a test if the user gives an incorrect offset?

Reply:
There is not, that's a good edge case! Added.